tests/test_FLEbasis2d.py

```diff
@@ -0,0 +1,274 @@
+
+        # cannot pass in the wrong number of coefficients
+        with self.assertRaisesRegex(
+            AssertionError, "Number of coefficients must match self.count."
+        ):
+            _ = basis.rotate(np.zeros((1, 10)), np.pi)
+
+        # cannot pass in wrong shape
+        with self.assertRaisesRegex(
+            AssertionError,
+            "Input a stack of coefficients of dimension",
+        ):
+            _ = basis.lowpass(np.zeros((3, 3, 3)), np.pi)
+
+    def testRadialConvolution(self):
+        # test ability to accurately convolve with a radial
+        # (e.g. CTF) function via FLE coefficients
+
+        L = 32
+        basis = FLEBasis2D(L)
+        # load test CTF
+        ctf = np.load(os.path.join(DATA_DIR, "ctf_32x32.npy"))
+        ctf = ctf / np.max(np.abs(ctf.flatten()))
+
+        # get sample images
+        ims = self.create_images(L, 10)
+        # convolve using coefficients
+        coeffs = basis.evaluate_t(ims)
+        coeffs_convolved = basis.radialconv(coeffs, ctf)
+        imgs_convolved_fle = basis.evaluate(coeffs_convolved).asnumpy()
+
+        # convolve using FFT
+        ctf = basis.evaluate(basis.evaluate_t(ctf)).asnumpy()
+        ims = basis.evaluate(coeffs).asnumpy()
+
+        imgs_convolved_slow = np.zeros((10, L, L))
+        for i in range(10):
+            ctf_pad = np.zeros((2 * L, 2 * L))
+            ims_pad = np.zeros((2 * L, 2 * L))
+            ctf_pad[L // 2 : L // 2 + L, L // 2 : L // 2 + L] = ctf[0, :, :]
+            ims_pad[L // 2 : L // 2 + L, L // 2 : L // 2 + L] = ims[i, :, :]
+
+            ctf_shift = fft.fftshift(ctf_pad.reshape(2 * L, 2 * L))
+            ims_shift = fft.fftshift(ims_pad.reshape(2 * L, 2 * L))
+
+            convolution_fft_pad = fft.fftshift(
+                fft.ifft2(np.fft.fft2(ctf_shift) * np.fft.fft2(ims_shift))
+            )
+            imgs_convolved_slow[i, :, :] = np.real(
+                convolution_fft_pad[L // 2 : L // 2 + L, L // 2 : L // 2 + L]
+            )
+
+        self.assertTrue(np.allclose(imgs_convolved_fle, imgs_convolved_slow, atol=1e-5))
+
+    def testFBCoefficients(self):
+        # Matching FB and FLE coefficients is an open issue
+        basis = FLEBasis2D(32, match_fb=True)
+        fb = FBBasis2D(32)
+
+        # The only thing we can guarantee for now is count
+        self.assertEqual(basis.count, fb.count)
+
+    def create_images(self, L, n):
+        # create sample data
+        v = Volume(
+            np.load(os.path.join(DATA_DIR, "clean70SRibosome_vol.npy")).astype(
+                np.float64
+            )
+        )
+        v = v.downsample(L)
+        sim = Simulation(L=L, n=n, vols=v, dtype=v.dtype, seed=1103)
+        img = sim.clean_images[:]
+        return img
+
+    def relerr(self, x, y):
+        # relative error of two arrays
+        x = np.array(x).flatten()
+        y = np.array(y).flatten()
+        return np.linalg.norm(x - y) / np.linalg.norm(x)
```

Comment:
I think it matters which value you expect to be more faithful. In this case, since you are dividing by `norm(x)` you are assuming that `y` is an approximation of `x`. Maybe this can be noted by more descriptive variable names.